File: frontend/src/types/order.ts
export type OrderStatus = 'New' | 'Approved' | 'Processing' | 'Shipped' | 'Delivered' | 'Cancelled'

export interface OrderItem {
  id: string
  productId: string
  productName: string
  price: number
  quantity: number
  lineTotal: number
}

export interface Order {
  id: string
  status: OrderStatus
  totalAmount: number
  shippingAddress: string
  contactPhone: string
  createdAt: string
  items: OrderItem[]
}

export interface AdminOrder extends Order {
  userId: string
  customerEmail: string
  customerFullName: string
  allowedNextStatuses: OrderStatus[]
}

File: backend/src/Store.Domain/Entities/Order.cs
using Store.Domain.Common;
using Store.Domain.Enums;

namespace Store.Domain.Entities;

public class Order : BaseEntity
{
    public Guid UserId { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.New;
    public decimal TotalAmount { get; set; }
    public string ShippingAddress { get; set; } = null!;
    public string ContactPhone { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
}
